astropy/io/fits/convenience.py

```diff
@@ -188,6 +188,7 @@ def getdata(filename, *args, **kwargs):
     data = hdu.data
     if data is None and extidx == 0:
         try:
+            hdulist, extidx = _getext(filename, mode, ext=1, *args, **kwargs)
```

Comment:
the extension 1 is already hardcoded, before it just loaded all and threw away all except the first one, see the line below